src/stdlib_math.fypp

```diff
@@ -1,19 +1,147 @@
+    real(dp), parameter :: DEFAULT_LOGSPACE_BASE = 10
+
+    ! Useful constants for lnspace 
+    real(sp), parameter :: EULERS_NUMBER_DP = 2.71828182845904523536028747135266249775724709369995_sp
+    real(dp), parameter :: EULERS_NUMBER_SP = 2.71828182845904523536028747135266249775724709369995_dp
+    real(qp), parameter :: EULERS_NUMBER_QP = 2.71828182845904523536028747135266249775724709369995_qp
 
     interface clip
     #:for k1, t1 in IR_KINDS_TYPES
         module procedure clip_${k1}$
     #:endfor
     end interface clip
 
+    interface linspace 
+    !! Version: Experimental
+    !!
+    !! Create rank 1 array of linearly spaced elements
+    !! If the number of elements is not specified, create an array with size 100. If n is a negative value,
+    !! return an array with size 0. If n = 1, return end
+    !!([Specification](../page/specs/stdlib_math.html#description))
+    #:for k1, t1 in RC_KINDS_TYPES
+      #:set RName = rname("linspace_default", 1, t1, k1)
+      module function ${RName}$(start, end) result(res)
+        ${t1}$, intent(in) :: start
+        ${t1}$, intent(in) :: end  
+  
+        ${t1}$ :: res(DEFAULT_LINSPACE_LENGTH)        
+      end function ${RName}$
+    #:endfor
+  
+    #:for k1, t1 in RC_KINDS_TYPES
+      #:set RName = rname("linspace_n", 1, t1, k1)
+      module function ${RName}$(start, end, n) result(res)
+        ${t1}$, intent(in) :: start
+        ${t1}$, intent(in) :: end  
+        integer, intent(in) :: n
+      
+        ${t1}$ :: res(n)
+      end function ${RName}$
+    #:endfor
+
+
+    ! Add suport for integer linspace    
+    !! Version: Experimental
+    !!
+    !! Create rank 1 array of linearly spaced elements from start to end.
+    !! If the number of elements is not specified, create an array with size 100. If n is a negative value,
+    !! return an array with size 0. If n = 1, return end. When dealing with integers as the `start` and `end` 
+    !! paramaters, the return type is always a double precision real.
+    !!
+    !!([Specification](../page/specs/stdlib_math.html#description))
+    #:for k1, t1 in INT_KINDS_TYPES
+      #:set RName = rname("linspace_default", 1, t1, k1)
+      #! The interface for INT_KINDS_TYPES cannot be combined with RC_KINDS_TYPES
+      #! because the output for integer types is always a real with dp.
+      module function ${RName}$(start, end) result(res)
+        ${t1}$, intent(in) :: start
+        ${t1}$, intent(in) :: end  
+  
+        real(dp) :: res(DEFAULT_LINSPACE_LENGTH)        
+      end function ${RName}$
+    #:endfor
+  
+    #:for k1, t1 in INT_KINDS_TYPES
+      #:set RName = rname("linspace_n", 1, t1, k1)
+      module function ${RName}$(start, end, n) result(res)
+        ${t1}$, intent(in) :: start
+        ${t1}$, intent(in) :: end  
+        integer, intent(in) :: n
+      
+        real(dp) :: res(n)
+      end function ${RName}$
+    #:endfor
+
+  end interface
+
+  interface logspace 
```

Comment:
Not in the specs, right?

Reply:
Not yet, I'm gonna get started working on the specs asap.